PR: https://github.com/odoo/odoo/pull/

From: 898224f110932addb0e976a3ca60542cda80de25
From: Deep Patel
Date: 2017-09-01 11:02:54

Breaking data model changes score: 3, change matches:
-    _inherit = "project.config.settings"
-    pad_server = fields.Char(related="company_id.pad_server", string="Pad Server")
-    pad_key = fields.Char(related="company_id.pad_key", string="Pad Api Key")

Total Changes: 12

[IMP] web,account,...: Regroup settings, add a nav and search bar

Purpose
=======

Settings are often way too long and hard to scan and sometimes you don't know where to find the settings you're looking for.
By adding a left navigation, you can already have an overview of the settings, and switch easily between them.

Specification
=============

* Add a search bar on all the apps that have settings which can search results from all the installed apps.
  (If i'm on sales settings,and I search anything then it shows results from sales settings and also shows
  all the other matching results from all the other apps such as, Inventory...etc) with app name.
* Able to activate feature from the current page results (no matter if the searched result is from another apps).
* Highlight searched word in results
* Delete the sheet, have a full white background
* Add left navigation bar on setting
* Left navigation bar fixed
* List displayed based on installed apps
* On right panel, by default display current app setting and change accordingly
* [Mobile] Left navigation bar displayed on top
* Add Breadcrumb on top of the page: have the name "Settings" + Save / discard CTA + Search
* Add General Settings on the nav bar
* remove "save this page..." notif in all settings
* Delete all recommanded apps section + all checkbox that install app should disappear:
	payroll: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrSkMzWVpuZ0ZoaFE/view?usp=drivesdk
	Events: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrYXlHMHd2NTM2blE/view?usp=drivesdk
	Manufacturing: Delete Repair - Quality Control - Maintenance - Product Lifecycle Management [LAP][ok]
	Timesheets: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrdFg5XzNydkd2TlU/view?usp=drivesdk
	project: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrraWo3NE04TktwcTQ/view?usp=drivesdk
	inventory: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrQkRaVUR5ekFQYTQ/view?usp=drivesdk
	recruitment: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrbDdqLWQweF80UkE/view?usp=drivesdk
	purchase: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrRlAyUDBnb0trQWs/view?usp=drivesdk
	email marketing: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrOUZONjhvX2k3Y2s/view?usp=drivesdk
	expenses: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrR0JpN1ZpQmtxU3M/view?usp=drivesdk
	attendances: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrralpjZ1VJclNpMWs/view?usp=drivesdk
* About duplicate settings:
	Docsaway: Delete from Sales + delete Default Print Provider
	Attributs & Variants: Delete from Purchases - Manufacturing
	Multi-currencies: delete from Sales
	Unit Of Measures: Delete from Purchase & Expenses
* Add a scroll bar on the left bar when there's too many apps
* Keep the navbar visible even when you search
* Move General Settings to the bottom of the list
* Update on Settings:
    - Accounting:
        - rename automatic rates
        - Anglo Saxon Account: Should be in technical feature
        - Place Accounting Reports section before Taxes section
    - CRM:
        - Phone Validation: Enforce international format becomes
        - Local Numbers: (2 radio buttons proposals) Add international prefix / No prefix
    - Fleet: typo " ... a new car if ..." + text is too long <br>
    - project: fix tooltip for colab pads

================================= pseudo patch: =================================

--- a/addons/pad_project/models/__init__.py
+++ b/addons/pad_project/models/__init__.py
@@ -1,5 +1,4 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from . import project_config_settings
 from . import project

--- a/addons/pad_project/models/project_config_settings.py
+++ b/None
@@ -1,11 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class ProjectConfigSettings(models.TransientModel):
-    _inherit = "project.config.settings"
-
-    pad_server = fields.Char(related="company_id.pad_server", string="Pad Server")
-    pad_key = fields.Char(related="company_id.pad_key", string="Pad Api Key")
